File: guides/micronaut-security-keys-jwks/java/src/main/java/example/micronaut/PrimarySignatureConfiguration.java
package example.micronaut;

import io.micronaut.runtime.context.scope.Refreshable;
import jakarta.inject.Named;

@Refreshable // <1>
@Named("generator") // <2>
public class PrimarySignatureConfiguration extends AbstractRSAGeneratorSignatureConfiguration {

    public PrimarySignatureConfiguration(JwkConfiguration jwkConfiguration) {
        super(jwkConfiguration.getPrimary());
    }
}

File: guides/micronaut-security-keys-jwks/kotlin/src/main/kotlin/example/micronaut/PrimarySignatureConfiguration.kt
package example.micronaut

import io.micronaut.runtime.context.scope.Refreshable
import jakarta.inject.Named

@Refreshable // <1>
@Named("generator") // <2>
class PrimarySignatureConfiguration(jwkConfiguration: JwkConfiguration) :
        AbstractRSAGeneratorSignatureConfiguration(jwkConfiguration.primary)
